LgiNetwork/Network.cpp

```diff
@@ -0,0 +1,1086 @@
+                    break;
+                }
+                i++;
+            }
+            return retval;
+	    }
+
+        // Wrapper methods
+        uint32_t Network::getQuirks(const JsonObject& parameters, JsonObject& response)
+        {
+            JsonArray array;
+            array.Add("RDK-20093");
+            response["quirks"] = array;
+            returnResponse(true)
+        }
+
+        uint32_t Network::getInterfaces (const JsonObject& parameters, JsonObject& response)
+        {
+            bool result = false;
+
+            if (m_isPluginInited)
+            {
+                JsonArray networkInterfaces;
+                std::vector<std::string>* interfaces = m_NetworkClient.getInterfaces();
+                if (interfaces)
+                {
+                    for (auto& iface : *interfaces)
+                    {
+                        JsonObject interface;
+                        std::map<std::string, std::string> parameters;
+                        parameters[lginet::ParamMac] = "";
+                        parameters[lginet::ParamConnectivity] = "";
+
+                        if (iface == "")
+                            continue;
+                        if (m_NetworkClient.getSpecificParamsForInterface(iface, parameters))
+                        {
+                            interface["interface"] = iface;
+                            interface["macAddress"] = parameters[lginet::ParamMac];
+                            interface["connected"] = (parameters[lginet::ParamConnectivity] == "yes");
+                            interface["enabled"] = m_NetworkClient.isInterfaceEnabled(iface);
+                            networkInterfaces.Add(interface);
+                        }
+                    }
+                    delete interfaces;
+                    response["interfaces"] = networkInterfaces;
+                    result = true;
+                }
+                else
+                {
+                    LOGWARN ("Call for %s failed", __FUNCTION__);
+                }
+            }
+            else
+            {
+                LOGWARN ("Network plugin not initialised yet returning from %s", __FUNCTION__);
+            }
+
+            returnResponse(result)
+        }
+
+        uint32_t Network::getDefaultInterface (const JsonObject& parameters, JsonObject& response)
+        {
+            string interface;
+            string gateway;
+            string type;
+
+            bool result = false;
+            if (m_isPluginInited)
+            {
+                if (_getDefaultInterface(interface, gateway, type))
+                {
+                    response["interface"] = type;
+                    result = true;
+                }
+                else
+                {
+                    LOGWARN ("Network plugin not initialised yet returning from %s", __FUNCTION__);
+                }
+
+                returnResponse(result)
+            }
+            return 0;
```

Comment:
It should return WPEFramework::Core::ERROR_*, probably WPEFramework::Core::ERROR_GENERAL here

Reply:
Changed